File: src/main/java/com/github/simiacryptus/aicoder/util/TextBlockFactory.java
package com.github.simiacryptus.aicoder.util;

import org.jetbrains.annotations.NotNull;

public interface TextBlockFactory<T extends TextBlock> {
    T fromString(String text);
    @SuppressWarnings("unused")
    default CharSequence toString(@NotNull T text) {
        return text.toString();
    }
    boolean looksLike(String text);
}

File: src/main/kotlin/com/simiacryptus/aicoder/util/TextBlockFactory.kt
package com.simiacryptus.aicoder.util

interface TextBlockFactory<T : TextBlock?> {
  fun fromString(text: String?): T

  @Suppress("unused")
  fun toString(text: T): CharSequence? {
    return text.toString()
  }

  fun looksLike(text: String?): Boolean
}
